api/src/opentrons/hardware_control/backends/ot3controller.py

```
@@ -933,6 +936,42 @@ def _axis_map_to_present_nodes(
         by_node = {axis_to_node(k): v for k, v in to_xform.items()}
         return {k: v for k, v in by_node.items() if k in self._motor_nodes()}
 
+    @asynccontextmanager
+    async def monitor_overpressure(
+        self, mount: OT3Mount, sensor_id: SensorId = SensorId.S0
+    ) -> AsyncIterator[None]:
+        if ff.overpressure_detection_enabled():
+            tool = sensor_node_for_pipette(OT3Mount(mount.value))
+            # FIXME we should switch the sensor type based on the channel
+            # used when partial tip pick up is implemented.
+            # FIXME we should also monitor pressure in all available channels
+            provided_context_manager = await check_overpressure(
+                self._messenger, tool, sensor_id
+            )
+            errors: asyncio.Queue[ErrorCode] = asyncio.Queue()
+
+            async with provided_context_manager() as errors:
+                try:
+                    yield
+                finally:
```

Not quite, I think. As implemented, we'll now correctly do whatever cleanup we need but we'll also swallow and blackhole the exception. We need to put the `_pop_queue` definition and call _inside_ the `finally` block so that it both runs even if there's an exception _and_ lets that exception bubble up afterward.